documentation/Generate-code-from-TypeSpec.md

```diff
@@ -0,0 +1,116 @@
+# Before you Start
+
+[TypeScript Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/typescript_introduction.html) is the overall design guideline of the client SDK.
+
+## Prerequisites
+
+- [Node.js 18.x LTS](https://nodejs.org/en/download) or later
+- [Git](https://git-scm.com/downloads)
+- [azure-rest-api-specs repo](https://github.com/Azure/azure-rest-api-specs)
+- [azure-sdk-for-js repo](https://github.com/Azure/azure-sdk-for-js/)
+
+# Generate SDK
+
+## Use TypeSpec defined in REST API specifications
+
+It is recommended to configure TypeSpec package on [REST API specifications](https://github.com/Azure/azure-rest-api-specs). Please refer to [these guidelines](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/typespec-structure-guidelines.md).
+
+### How to configure tspconfig.yaml
+You can reference these two config files to configure the Modular or RLC package:
+- [Modular tspconfig.yaml](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml)
+- [RLC tspconfig.yaml](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/ai/Face/tspconfig.yaml)
+
+Please make sure `service-dir`, `package-dir`, `packageDetails`, `isModularLibrary`, `generateMetadata`, `flavor`(for typespec-ts) is correctly configured. `experimentalExtensibleEnums`, `enableOperationGroup`, `hierarchyClient` are the optional configs.
```

Comment:
Do we have any place to describe the meaning of those configurations?

Reply:
Looks like Go doc also provide samples only. I think @maririos is working on describing options for all languages, when it's done, we can add the link